state/store.go

```diff
@@ -366,6 +389,51 @@ func (store dbStore) PruneStates(from int64, to int64, evidenceThresholdHeight i
 	return nil
 }
 
+// PruneABCIResponses attempts to prune all ABCI responses up to, but not
+// including, the given height. On success, returns the height to which ABCI
+// responses were pruned.
+func (store dbStore) PruneABCIResponses(height int64) (int64, int64, error) {
+	if store.DiscardABCIResponses {
+		return 0, 0, errors.New("ABCI responses are discarded, nothing to prune")
+	}
+	lastPruneHeight, err := store.getLastABCIResponsesPruneHeight()
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to look up last ABCI responses prune height: %w", err)
+	}
+	if lastPruneHeight == 0 {
+		lastPruneHeight = 1
+	}
+
+	batch := store.db.NewBatch()
+	defer batch.Close()
+
+	pruned := int64(0)
+	batchPruned := int64(0)
+
+	for h := lastPruneHeight; h < height; h++ {
+		if err := batch.Delete(calcABCIResponsesKey(h)); err != nil {
+			return pruned, lastPruneHeight + pruned, fmt.Errorf("failed to delete ABCI responses at height %d: %w", h, err)
+		}
+		batchPruned++
+		if batchPruned%1000 == 0 && batchPruned > 0 {
+			if err := batch.Write(); err != nil {
+				return pruned, lastPruneHeight + pruned, fmt.Errorf("failed to write ABCI responses deletion batch at height %d: %w", h, err)
+			}
+			batch.Close()
+
+			pruned += batchPruned
+			batchPruned = 0
+			if err := store.setLastABCIResponsesPruneHeight(h); err != nil {
+				return pruned, lastPruneHeight + pruned, fmt.Errorf("failed to set last ABCI responses prune height: %w", err)
+			}
+
+			batch = store.db.NewBatch()
+			defer batch.Close()
+		}
+	}
```

Comment:
I forgot to add this previously:

```suggestion
	}
	if err := store.setLastABCIResponsesPruneHeight(height); err != nil {
		return pruned, height, fmt.Errorf("failed to set last ABCI responses prune height: %w", err)
	}
```